sotodlib/site_pipeline/calibration/wiregrid.py

```diff
@@ -38,61 +92,76 @@ def _wrap_wg_hk(tod, telescope=None, hk_dir=None):
+                - LSL2 : ON/OFF status of the limit switch LEFT 2 (inside) of the actuator
+                - LSR1 : ON/OFF status of the limit switch RIGHT 1 (outside) of the actuator
+                - LSR2 : ON/OFF status of the limit switch RIGHT 2 (inside) of the actuator
 
     """
-    if telescope is None:
-        try:
-            telescope = tod.obs_info.telescope
-        except:
-            print("Cannot load the telescope info. Please assign your target telescope")
-            logger.info("Telescope assignment was failed in site_pipeline.calibrate.wiregrid._wrap_wg_hk")
-    #
-    alias_dict = {
-        'enc_rad_raw': telescope + '.wg-encoder.feeds.wgencoder_full.reference_count',
-        'limitswitchL1': telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSL1',
-        'limitswitchL2': telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSL2',
-        'limitswitchR1': telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSR1',
-        'limitswitchR2': telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSR2',
-        }
-    if hk_dir is None: hk_dir = '/so/level2-daq/' + telescope + '/hk'
-    #
-    _gl_aman = hk_utils.get_detcosamp_hkaman(tod,
-                                     alias = [_i for _i in alias_dict.keys()],
-                                     fields = [_j for _j in alias_dict.values()],
-                                     data_dir = hk_dir)
-    _gl_data = {}
-    for _i in alias_dict.keys():
-        if _i == 'enc_rad_raw':
-            _gl_data[_i] = _gl_aman['wg-encoder']['wg-encoder'][0]
-        else:
-            _gl_data[_i] = _gl_aman.restrict('hklabels_wg-actuator', [_i], in_place=False)['wg-actuator']['wg-actuator'][0]
-    #
-    _ax_wg = core.AxisManager(tod.samps, tod.dets)
-    _ax_wg.wrap('enc_rad_raw', _gl_data['enc_rad_raw'] * wg_counts2rad, [(0,'samps')])
-    _ax_wg.wrap('LSL1', _gl_data['limitswitchL1'], [(0,'samps')])
-    _ax_wg.wrap('LSL2', _gl_data['limitswitchL2'], [(0,'samps')])
-    _ax_wg.wrap('LSR1', _gl_data['limitswitchR1'], [(0,'samps')])
-    _ax_wg.wrap('LSR2', _gl_data['limitswitchR2'], [(0,'samps')])
-    tod.wrap('wg', _ax_wg)
+    tod_start = float(tod.obs_info.start_time) - ts_margin
+    tod_stop =  float(tod.obs_info.stop_time) + ts_margin
+
+    # Define house-keepind data configurations
+    _tel = tod.obs_info.telescope
+    hk_dir = '/so/level2-daq/' + _tel + '/hk'
+    wg_encoder_fields = {
+            'enc_rad_raw': _tel + '.wg-encoder.feeds.wgencoder_full.reference_count',
+    }
+    wg_actuator_fields = {
+        'LSL1': _tel + '.wg-actuator.feeds.wgactuator.limitswitch_LSL1',
+        'LSL2': _tel + '.wg-actuator.feeds.wgactuator.limitswitch_LSL2',
+        'LSR1': _tel + '.wg-actuator.feeds.wgactuator.limitswitch_LSR1',
+        'LSR2': _tel + '.wg-actuator.feeds.wgactuator.limitswitch_LSR2',
+    }
+
+    try:
+        # for encoder data
+        _data_enc = load_range(
```

Comment:
Temporarily, I'll prepare the two kinds of wrappers. I'll leave the hkdb one as just a place holder for a while because the method is not available in NERSC yet. Somebody or probably I will replace it in the next pull request.